src/sinks/aws_s3.rs

```diff
@@ -132,15 +128,26 @@ impl S3Sink {
 
         let s3 = S3Sink {
             client: Self::create_client(config.region.clone().try_into()?),
-            bucket: config.bucket.clone(),
-            gzip: compression,
-            filename_time_format,
-            filename_append_uuid,
-            filename_extension: config.filename_extension.clone(),
         };
 
-        let sink = request
-            .batch_sink(S3RetryLogic, s3, acker)
+        let filename_extension = config.filename_extension.clone();
+        let bucket = config.bucket.clone();
+
+        let svc = ServiceBuilder::new()
+            .map(move |req| {
+                build_request(
+                    req,
+                    filename_time_format.clone(),
+                    filename_extension.clone(),
+                    filename_append_uuid,
+                    compression,
+                    bucket.clone(),
+                )
+            })
+            .settings(request, S3RetryLogic)
```

Comment:
It's not very clear what `settings` means here (`request` is also a bad name, but I know that was there before).

Reply:
So settings here relate to `TowerRequestSettings` struct, which I also am not a big fan of that name but I am trying to tie it to that.